content/the-basics-of-go.md

```diff
@@ -0,0 +1,864 @@
+---
+title: The Basics of Go
+desc: Learn the basics of Golang in just one single article
+author: Reinaldy Rafli
+github: aldy505
+twitter:
+telegram: aldy505
+date: 2021-11-09
+cover: https://www.jeremymorgan.com/images/learn-golang/learn-golang-og.jpg
+categories:
+  - golang
+---
+
+First and foremost, I can't teach this material in Indonesian. So, to understand this article, you should know
+at least basic of English. Even if you don't, you can always try Google Translate.
+
+## Short introduction
+
+Go (usually written as Go, but pronounced as Golang) came around in the year of 2009 by Google. It's a compiled,
+statically-typed language where unlike Python or Javascript, it directly compiles to machine code and have strict
+type declaration for variables and functions.
+
+A few things that I like from Go are:
+
+1. Very easy to install, and its toolings are available for most text editors, even IDE.
+2. Very lightweight when compiled to binary (yes, it compiles to a binary format that you can run anywhere)
+3. Boring code, meaning the code, whoever it is that wrote it, would be easy to read and understand by anyone.
+4. The standard libraries are not as confusing as Node.js or Rust's.
+5. Automatic documentation generation for packages.
+6. Can be easily decoupled, you can put a single packagee and take out another package without having headaches.
+
+## Some things to be acknowledged
+
+Like all programming language, the first thing that you should do is to drop how to do things on any other
+programming language, but just keep the concept of it.
+
+In Go, most people from Java and PHP might say, "How can this language be good? It doesn't have class keyword,
+it can't do OOP!". Well, you're wrong. OOP concepts such as inheritance, encapsulations and others don't really rely
+on the class keyword itself. You can do OOP in Go, but do you really need OOP tho? There are lots and lots of other
```

Comment:
Telling about Go's OOP work in [composition manner](https://golangbot.com/inheritance/), m8 be good.

Fun-fact: In an alien language from λ-solar system such Haskell, OCaml and etc, this approach are heavily used but with different name for different paradigm : Typeclass or Functor.

Reply:
no

Reply:
oke resolved